### PR TITLE
Add several tests to templates

### PR DESCRIPTION
**Pull request by cezaraugusto**
Added a bunch of tests to cover CLI commands against the base templates.

Files changed ended up too big but mostly due to the fixtures I had to add to test the build command.

I also made a few changes to templates and tried to come up with a pattern for new additions. This is a starter to allow authors to add their own templates to our list.

```
# uiContext = 'sidebar' | 'newTab' | 'content' | 'popup' | 'devTools'

[template]/
├── images/
│   └── icons/
│       ├── icon_16.png
│       └── icon_48.png
├── [uiContext]/
│   ├── index.html        # (only for non content_scripts)
│   ├── styles.css
│   ├── scripts.js        # (or .jsx or .ts or .tsx)
│   └── [uiContextApp].ts # (or .jsx or .ts or .tsx – only for framework-based templates)
├── env.example           # (optional)
├── manifest.json
├── postcss.config.js     # (for tailwind templates)
├── README.md
├── tailwind.config.js    # (for tailwind templates)
└── .gitignore
```

Feedback welcome!

### CODE REVIEW

File: programs/cli/cli.ts
Comment (OSpoon) [inline programs/cli/cli.ts:1]: The statement that imports `semver` can be deleted.

Comment (OSpoon) [inline programs/cli/cli.ts:1]: This code refactoring is very good.

File: programs/cli/package.json
Comment (OSpoon) [inline programs/cli/package.json:1]: I would like to remind you here that `before:test` seems to run only in non-win environment at present, which may be irrelevant.

**Review by OSpoon — COMMENTED** [review]
I have confirmed these unit tests and found no issues other than they run slowly.

Do you have any thoughts on ensuring the extension operates correctly in a browser environment? Should we consider incorporating end-to-end testing for this purpose?

File: programs/cli/cli.ts
Comment (cezaraugusto) [reply programs/cli/cli.ts:1]: got it, deleted

**Comment by cezaraugusto** [comment]
@OSpoon thanks for reviewing!

Regarding the e2e tests: yes, that would be ideal. This PR adds tests for `create` and `build` commands only, so having `dev` and `start` command tests against the default templates will help us a lot to prevent some bugs similar to those we are currently fixing

**Review by OSpoon — APPROVED** [review]
LGTM++

**Pull request merged**